lms/djangoapps/courseware/module_render.py

```diff
@@ -713,7 +713,6 @@ def rebind_noauth_module_to_user(module, real_user):
         wrappers=block_wrappers,
         get_real_user=user_by_anonymous_id,
         services={
-            'i18n': ModuleI18nService(),
             'fs': FSService(),
             'field-data': field_data,
             'user': DjangoXBlockUserService(user, user_is_staff=user_is_staff),
```

Comment:
@cpennington, is this what you had in mind?  @asadiqbal08, does this new
test fail if you remove the ModuleI18nService() call above?
On Mar 9, 2016 7:45 AM, "Asad Iqbal" <notifications@github.com> wrote:

> In lms/djangoapps/courseware/module_render.py
> <https://github.com/edx/edx-platform/pull/11575#discussion_r55512583>:
>
> > @@ -713,7 +713,6 @@ def rebind_noauth_module_to_user(module, real_user):
> >          wrappers=block_wrappers,
> >          get_real_user=user_by_anonymous_id,
> >          services={
> > -            'i18n': ModuleI18nService(),
> >              'fs': FSService(),
> >              'field-data': field_data,
> >              'user': DjangoXBlockUserService(user, user_is_staff=user_is_staff),
>
> @mattdrayer <https://github.com/mattdrayer> is that you required to test
> ?
>
>
> def assert_services(self, module):
>         """
>         assert possible services
>         """
>         self.assertTrue(module.runtime._services.get('i18n', False))
>         self.assertTrue(module.runtime._services.get('fs', False))
>         self.assertTrue(module.runtime._services.get('user', False))
>         self.assertTrue(module.runtime._services.get('reverification', False))
>         self.assertTrue(module.runtime._services.get('proctoring', False))
>         self.assertTrue(module.runtime._services.get('credit', False))
>         self.assertTrue(module.runtime._services.get('bookmarks', False))
>
> def test_services_before_and_after_module_rebinding(self):
>         """
>         LTI XBlock that calls the re-bind method and confirms the proper services.
>         """
>         module = self.get_module_for_user(self.anon_user)
>         # assert services before rebinding
>         self.assert_services(module)
>         user2 = UserFactory.create()
>         module.system.rebind_noauth_module_to_user(module, user2)
>         # assert services after rebinding
>         self.assert_services(self.get_module_for_user(user2))
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/11575/files#r55512583>.
>